src/bikeracks/templates/scripts.js

```diff
@@ -43,7 +43,11 @@ class BikeMap {
         this.$signInButton.click(this.signIn.bind(this));
         
         // placing temporary marker on map event listeners
-        this.mymap.on('contextmenu', this.onMapClick.bind(this));
+        this.mymap.on('contextmenu', this.onRightClick.bind(this));
+        // remove temporary marker when clicking on map
+        this.mymap.on('click', function() {
+            this.mymap.removeLayer(this.tempMarker);
```

Comment:
Does this function need to be in parens to call `.bind()` on it?

Reply:
I don't think so. I've written all of my functions this way. Also, I just tested it in the console and it seems to work.